src/tools/edit_tool.h

```diff
@@ -144,10 +145,28 @@ Q_OBJECT
 	 */
 	void drawBoundingPath(QPainter* painter, MapWidget* widget, const std::vector<QPointF>& bounding_path, const QRgb& color);
 	
+	/**
+	 * Rendering of some of the editor graphic elements may depend on
+	 * MapView settings. This method is the place where to process
+	 * the changes.
+	 *
+	 * @param change type of change as emitted by MapView
+	 */
+	virtual void applyViewChangesImpl(MapView::ChangeFlags change) = 0;
+
 	/**
 	 * An utility implementing object selection logic.
 	 */
 	QScopedPointer<ObjectSelector> object_selector;
+
+public slots:
+	/**
+	 * This slot forwards viewChange notifications from MapView to the
+	 * sub-class specific implementation.
+	 * 
+	 * @param change 
+	 */
+	void applyViewChanges(OpenOrienteering::MapView::ChangeFlags change);
```

Comment:
I'd prefer to make this slot a virtual method, with a default implementation which does nothing ATM, and with an explanation that overriding implementations shall call the parent's implementation.

Although common in Mapper, it is sometimes not really clear what problem is solved by the "impl" approach. I suppose the common real use case is when you want to have efficient variation in some part of an algorithm. This is where Qt uses virtual "event" methods (e.g. paintEvent). But having slots and (separate) virtual methods appears to be a lot of indirection.

Reply:
Right. A virtual method was one of my first thoughts but for some reason, I thought it cannot be used as a slot. It turns out that it was just my careless reading of the Signals & Slots chapter in Qt documentation, which explicitly mentions this option. I will fix this part.